htlcswitch/quiescer.go

```diff
@@ -233,3 +236,23 @@ func (q *quiescer) initStfu(req StfuReq) {
 	q.localInit = true
 	q.activeQuiescenceReq = fn.Some(req)
 }
+
+// onResume accepts a no return closure that will run when the quiescer is
+// resumed.
+func (q *quiescer) onResume(hook func()) {
+	q.resumeQueue = append(q.resumeQueue, hook)
```

Comment:
Are we ever expecting to get 2x `RevokeAndAck` message while we're busy quiescing? Wondering whether we need a queue of remove hooks vs a single function. 

Mostly leaving this comment as a note to self to improve my own understanding!

Reply:
We shouldn't get multiple R&As but it's possible we may not want to bounce packets when quiescent in the future, we may want to queue them for when the channel goes live again. We could actually do away with this entirely if we move exit hop processing into the switch. It shouldn't be in the link anyway imo. It's an architectural issue I've pointed out on several occations. What should happen is that rather than processing the forwarding package inside the link we eject it into the switch and then if the switch notices that we can settle then it can drop a settle packet into the link's queue and the normal traffic control will handle it as opposed to needing to make a carve out here.

Reply:
> We shouldn't get multiple R&As but it's possible we may not want to bounce packets when quiescent in the future, we may want to queue them for when the channel goes live again

Maybe it's worth keeping it simple and being stricter with what we accept in this first iteration if we don't have a near term requirement for resuming? 

> See https://github.com/lightningnetwork/lnd/issues/8634

Will check it out!

Reply:
We do have a near-ish term requirement to resume though since completion of DynComms/Splicing will need to resume.

Reply:
> near-ish term requirement to resume though since completion of DynComms/Splicing

near-ish, heavy on the "ish" for the ~year of eng work till we have splicing :')

Would like to push for just having a single function and erroring out if we try `onResume` once, because that means we've got bugz / something seriously unexpected has happened. Diff to change it to `[]func` will be pretty small so I don't think it's high cost to start strict.